starcheck/src/lib/Ska/Parse_CM_File.pm

```diff
@@ -103,17 +93,33 @@ sub dither {
 	    }
           }
 	$dith_hist_fh->close();
-    
-    my @ok = grep { $dh_time[$_] < $bs_arr->[0]->{time} } (0 .. $#dh_time);
-    my @state = (@dh_state[@ok], @bs_state);
-    my @time   = (@dh_time[@ok], @bs_time);
-    my @params = (@dh_params[@ok], @bs_params);
+
+    # Confirm that last state matches kadi continuity ENAB/DISA
+    if ($kadi_dither->{'dither'} ne $dh_state[-1]){
+        print STDERR "Dither status in kadi commands does not match DITHER history\n";
+        return (1, undef);
+    }
+
+    my @state;
+    my @time;
+    my @params;
+    # Then get everything from backstop
+    foreach $bs (@{$bs_arr}) {
+	if ($bs->{cmd} =~ '(COMMAND_SW|MP_DITHER)') {
+	    my %params = %{$bs->{command}};
+	    if ($params{TLMSID} =~ 'AO(DSDITH|ENDITH|DITPAR)') {
+		push @state, $dith_cmd{$1};
```

Comment:
`@state` is a particularly confusing variable name.  This is really the dither enable/disable status, whereas "state" normally means a collection of correlated state values at a time (which is `@dither` later).  I think you could call it `dither_enab`, and rename `dither_cmd` to `dither_enab_cmd_map`.